src/_P003_Pulse.ino

```diff
@@ -194,7 +194,7 @@ boolean Plugin_003(byte function, struct EventStruct *event, String& string)
           event->sensorType                = Sensor_VType::SENSOR_TYPE_TRIPLE;
           UserVar[event->BaseVarIndex]     = Plugin_003_pulseCounter[event->TaskIndex];
           UserVar[event->BaseVarIndex + 1] = Plugin_003_pulseTotalCounter[event->TaskIndex];
-          UserVar[event->BaseVarIndex + 2] = Plugin_003_pulseTime[event->TaskIndex]/1000.0f;
+          UserVar[event->BaseVarIndex + 2] = (unsigned long)(Plugin_003_pulseTime[event->TaskIndex]/1000.0f);
```

Comment:
Why cast it to unsigned long?
You wanted to increase resolution, but now you cast it to msec again.

Reply:
That's only the UserVar (which I assumed is 32bit) that stores the PulseTime (in ms anyways, for compatibility reasons,, therefore `/1000.0f`), so there the overflow will (still) happen as before on the `milis()` version (some 50days).

But for the internal calculation of the difference 64bit values are used everywhere and calculations are don in µs.

That's just IMHO. though...

Reply:
You convert it to msec by dividing with 1000.0
but if you cast it then to an integer type, you will loose the gained resolution, so there is no extra resolution anymore from when we started this PR.

`millis()` (e.g. 123 ms) -> `micros() / 1000.0` (e.g. 123.456 ms) -> cast to int -> millis resolution again. (e.g. 123 ms)

Sure by using a 64 bit counter the 49.7 days bug is now fixed, but you're throwing away the microsec resolution.

Reply:
of course you're right (as always ;) this probably explains the strange issue I had.... this should be casted to a float I guess, which was my intention but I just messed up... will correct it!

Reply:
When you assign it to the UserVar, it will automatically be casted to float, as it is of type float.

Reply:
ok, thanks. So you would suggest to skip it? could leave it to make it clear, that these are different types, but it's your project ;) I'll adhere to your guidelines...

Reply:
done. also added `getMicros()` in commit #61e05159f9a036a9568e80bdb8a3964443044cbd